components/raftstore/src/store/peer.rs

```diff
@@ -2336,11 +2338,32 @@ where
             }
             Ok(RequestPolicy::ReadIndex) => return self.read_index(ctx, req, err_resp, cb),
             Ok(RequestPolicy::ProposeNormal) => {
-                let store_id = ctx.store_id();
-                if disk::disk_full_precheck(store_id) || ctx.is_disk_full {
-                    Err(Error::Timeout("disk full".to_owned()))
-                } else {
+                let mut allowed = true;
+                let mut err_msg = String::from("propose failed: disk ");
```

Comment:
It can be allocated when necessary. By the way, the error message seems not complete.

Reply:
It's not necessary to make 2 different error message. Just `propose failed: disk full` is OK for them.

Reply:
err_msg left will be filled in under each threshold.

Reply:
Done